### PR TITLE
Fix incremental compilation failing if a jar on classpath doesn't exist

### PR DESCRIPTION
**Pull request by melix**
Fixes: gradle/gradle#1457

### CODE REVIEW

File: subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/jar/JarClasspathSnapshotFactory.java [hunk 1]
```diff
@@ -39,12 +39,14 @@ JarClasspathSnapshot createSnapshot(Iterable<JarArchive> jarArchives) {
         Set<String> duplicateClasses = Sets.newHashSet();
 
         for (JarArchive jar : jarArchives) {
-            JarSnapshot snapshot = jarSnapshotter.createSnapshot(jar);
-            jarSnapshots.put(jar.file, snapshot);
-            jarHashes.put(jar.file, snapshot.getHash());
-            for (String c : snapshot.getClasses()) {
-                if (!allClasses.add(c)) {
-                    duplicateClasses.add(c);
+            if (jar.file.exists()) {
```
Comment (oehme) [inline subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/jar/JarClasspathSnapshotFactory.java]: Not sure if this is the right place or whether this filtering should be done where this method is called.

**Comment by melix** [comment]
Should we target 3.4.1 or 3.5 for this one?

**Comment by adammurdoch** [comment]
I'd include it in 3.4.1

**Comment by melix** [comment]
Merged into `release` instead.

**Pull request closed without merging**